utils/firestore.js

```diff
@@ -1410,137 +1410,72 @@ const getSpecimenCollections = async (token, siteCode) => {
+    const endDate = filters.endDate ?? '0';
+    try {
+        let query = db.collection('boxes').where('145971562', '==', 353358909);
 
-    let startDate = "0";
-    let trackingId = '';
-    let endDate = "0";
-    if(filters.hasOwnProperty('startDate')){
-        startDate = filters['startDate']
-    }
-    
-    if(filters.hasOwnProperty('trackingId')){
-        trackingId = filters['trackingId'];
-    }
-    if(filters.hasOwnProperty('endDate')){
-        endDate = filters['endDate']
-    }
-    let snapshot = {'docs':[]};
-    if(trackingId !== ''){ 
-        if(endDate !== "0"){ 
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).where('656548982', '<=', endDate).where('656548982', '>=', startDate).orderBy('656548982', 'desc').get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).where('656548982', '<=', endDate).orderBy('656548982', 'desc').get();
-            }
-      }
-      else{
-          if(startDate !== "0"){
-              snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).where('656548982', '>=', startDate).orderBy('656548982', 'desc').get();
-          }
-          else{
-              snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).orderBy('656548982', 'desc').get();
-          }
-      }
-    }
-    else{
-        if(endDate !== "0"){
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', '<=', endDate).where('656548982', '>=', startDate).orderBy('656548982', 'desc').get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', '<=', endDate).orderBy('656548982', 'desc').get();
-            }
-        }
-        else{
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', '>=', startDate).orderBy('656548982', 'desc').get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).orderBy('656548982', 'desc').get();
-            }
-        }
+        query = buildQueryWithFilters(query, trackingId, endDate, startDate, source, siteCode)
+
+        const snapshot = await query.orderBy('656548982', 'desc').get();
+        const result = snapshot.docs.length;
```

Comment:
```suggestion
        const snapshot = await query.count().get();
        const result = snapshot.data().count;
```

Reply:
As Joe mentioned, there's no need to pull all data (slow and costly), when we need only the count as an integer.

Reply:
Warren I don't think this going to work. 
```
const snapshot = await query.count().get();
const result = snapshot.data().count;
```
this is Web Version 9